migration/migration.go

```diff
@@ -0,0 +1,250 @@
+	if *nextVersion >= int64(len(m)) {
+		// No further updates to apply (this is NOT an error)
+		log.Info(nil, map[string]interface{}{
+			"next_version":    *nextVersion,
+			"current_version": currentVersion,
+		}, "Current version %d. Nothing to update.", currentVersion)
+		return nil
+	}
+
+	log.Info(nil, map[string]interface{}{
+		"next_version":    *nextVersion,
+		"current_version": currentVersion,
+	}, "Attempt to update DB to version %v", *nextVersion)
+
+	// Apply all the updates of the next version
+	for j := range m[*nextVersion] {
+		if err := m[*nextVersion][j](tx); err != nil {
+			return errs.Errorf("failed to execute migration of step %d of version %d: %s\n", j, *nextVersion, err)
+		}
+	}
+
+	if _, err := tx.Exec("INSERT INTO version(version) VALUES($1)", *nextVersion); err != nil {
+		return errs.Errorf("failed to update DB to version %d: %s\n", *nextVersion, err)
+	}
+
+	log.Info(nil, map[string]interface{}{
+		"next_version":    *nextVersion,
+		"current_version": currentVersion,
+	}, "Successfully updated DB to version %v", *nextVersion)
+
+	return nil
+}
+
+// getCurrentVersion returns the highest version from the version
+// table or -1 if that table does not exist.
+//
+// Returning -1 simplifies the logic of the migration process because
+// the next version is always the current version + 1 which results
+// in -1 + 1 = 0 which is exactly what we want as the first version.
+func getCurrentVersion(db *sql.Tx, catalog string) (int64, error) {
+	query := `SELECT EXISTS(
+				SELECT 1 FROM information_schema.tables
+				WHERE table_catalog=$1
+				AND table_name='version')`
+	row := db.QueryRow(query, catalog)
+
+	var exists bool
+	if err := row.Scan(&exists); err != nil {
+		return -1, errs.Errorf(`failed to scan if table "version" exists: %s\n`, err)
+	}
+
+	if !exists {
+		// table doesn't exist
+		return -1, nil
+	}
+
+	row = db.QueryRow("SELECT max(version) as current FROM version")
+
+	var current int64 = -1
+	if err := row.Scan(&current); err != nil {
+		return -1, errs.Errorf(`failed to scan max version in table "version": %s\n`, err)
+	}
+
+	return current, nil
+}
+
+// NewMigrationContext aims to create a new goa context where to initialize the
+// request and req_id context keys.
+// NOTE: We need this function to initialize the goa.ContextRequest
+func NewMigrationContext(ctx context.Context) context.Context {
+	req := &http.Request{Host: "localhost"}
+	params := url.Values{}
+	ctx = goa.NewContext(ctx, nil, req, params)
+	// set a random request ID for the context
+	var reqID string
+	ctx, reqID = client.ContextWithRequestID(ctx)
+
+	log.Debug(ctx, nil, "Initialized the migration context with Request ID: %v", reqID)
+
+	return ctx
+}
+
+// GetMigrations returns the migrations all the migrations we have.
+func getMigrations(migrateData MigrateData) Migrations {
+	m := Migrations{}
+	for _, file := range migrateData.AssetNames() {
+		m = append(m, steps{executeSQLFile(migrateData.Asset, file)})
+	}
+	return m
+}
```

Comment:
`getMigrations()` is about iterate `migrateData.AssetNames()` and call `executeSQLFile()` for each.